### PR TITLE
feat: add `lookupDomains` method to return an address owned ENS domains

### PR DESCRIPTION
**Pull request by wa0x6e**
This PR add a new API method `lookupDomains` to return an address owned ENS domains

The goal is to move https://github.com/snapshot-labs/snapshot/blob/1b1c37021a53470512c840a69d6b31c51068b912/src/composables/useEns.ts#L8 here to stamp, so that we only have one source of truth for a controller owned domains.

After this PR is merged

- we should refactor snapshot UI to fetch list of owned ENS from stamp (we keep the invalid domains validation on UI side)
- Add a new `controller` filter to Space on the hub, so that SX could return all spaces controlled by an address (like on SX API)

Only 2 files are changed (api.ts and lookupDomains.ts), as well as tests, the rest are refactoring because we moved a function

### Test

Run 

```shell
curl -X POST http://localhost:3008/ -H "Content-Type: application/json" -d '{"method": "lookup_domains", "params": "0x24F15402C6Bb870554489b2fd2049A85d75B982f" }'
```

It should return 

```json
{"jsonrpc":"2.0","result":["chaitu.eth","validation.snapspace.eth","snapspace.eth","2.snapspace.eth","1.snapspace.eth","thanku.eth","3.snapspace.eth","4.snapspace.eth"],"id":null}
```

### TODO 

- [x] Add tests

### CODE REVIEW

File: src/lookupDomains.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import { FetchError, isSilencedError } from './addressResolvers/utils';
+import { Address, graphQlCall } from './utils';
+import { isAddress } from '@ethersproject/address';
+
+const ENS_GRAPHQL_URL = 'https://api.thegraph.com/subgraphs/name/ensdomains/ens';
```
Comment (ChaituVR) [inline src/lookupDomains.ts:6]: Could we allow network as input? to get domains owned on sepholia for testnet spaces? not needed for now?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/lookupDomains.ts (hunk 1 above)
Comment (wa0x6e) [reply src/lookupDomains.ts:6]: Good idea

File: src/lookupDomains.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    ...domain,
+    name: labelName ? domain.name.replace(`[${hash}]`, labelName) : domain.name
+  };
+}
+
+export default async function lookupDomains(
+  address: Address,
+  chainId = DEFAULT_CHAIN_ID
+): Promise<Address[]> {
+  if (!isAddress(address) || !ENS_GRAPHQL_URL[chainId]) return [];
```
Comment (ChaituVR) [inline src/lookupDomains.ts:51]: Maybe better to return error?

**Review by ChaituVR — APPROVED** [review]
tAck

File: src/lookupDomains.ts (hunk 2 above)
Comment (wa0x6e) [reply src/lookupDomains.ts:51]: Maybe, but all the other methods do not return any errors, and just return `[]` instead.

Comment (ChaituVR) [reply src/lookupDomains.ts:51]: i see.. ok ok 👍

**Pull request merged**